### PR TITLE
Introduce new `botkube.io` website based on Docusaurus

### PR DESCRIPTION
**Pull request by madebyrogal**
## Description
Changes proposed in this pull request:
- implement botkube.io page using docusaurus instead of hugo

## Related issue(s)
#80 

## Preview
https://botkube-docs-c3c.pages.dev/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    "clear": "docusaurus clear",
+    "serve": "docusaurus serve",
+    "write-translations": "docusaurus write-translations",
+    "write-heading-ids": "docusaurus write-heading-ids",
+    "tsc": "tsc",
+    "lint": "npm run eslint:check && npm run prettier:check",
+    "eslint:check": "eslint . --ext .json,.js,.jsx,.ts,.tsx",
+    "prettier:check": "prettier --check .",
+    "prettier:fix": "prettier --write .",
+    "qa": "npm run tsc && npm run lint && npm run build"
```
Comment (pkosiec) [inline package.json:20]: You can consider different name, e.g. `verify`:
```suggestion
    "verify": "npm run tsc && npm run lint && npm run build"
```

You can ignore this if you don't agree 👍

File: package.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "docusaurus-website",
+  "version": "0.0.0",
```
Comment (pkosiec) [inline package.json:3]: Let's set at least `0.1.0` 😄 We don't need to change it later.
```suggestion
  "version": "0.1.0",
```

File: .github/workflows/dead-link-checker.yaml [hunk 3]
```diff
@@ -21,22 +21,14 @@ jobs:
...
         uses: actions/checkout@v3
-      - uses: actions/cache@v2
+      - name: Setup Node
+        uses: actions/setup-node@v3
         with:
-          path: /tmp/hugo_cache
-          key: ${{ runner.os }}-hugomod-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-hugomod-
+          node-version: 16.16
```
Comment (pkosiec) [inline .github/workflows/dead-link-checker.yaml:27]: 1. Enable cache
2. Use node version from file (https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file)

```suggestion
          node-version-file: '.nvmrc'
          cache: 'npm'
```

File: .github/workflows/qa-checker.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.16
```
Comment (pkosiec) [inline .github/workflows/qa-checker.yaml:23]: Same as before: 
```suggestion
          node-version-file: '.nvmrc'
          cache: 'npm'
```

File: package.json (hunk 1 above)
Comment (madebyrogal) [reply package.json:20]: I like this quality assurance shortcut ;) I would like to leave it as it is.

File: .github/workflows/dead-link-checker.yaml (hunk 3 above)
Comment (madebyrogal) [reply .github/workflows/dead-link-checker.yaml:27]: Ooo I didn't know about `node-version-file` thanks for that.
I've changed it also in another job (qa-checker)

File: .github/workflows/dead-link-checker.yaml [hunk 5]
```diff
@@ -21,22 +21,14 @@ jobs:
...
       - name: Setup Go
         uses: actions/setup-go@v3
         with:
           go-version: 1.18
```
Comment (huseyinbabal) [inline .github/workflows/dead-link-checker.yaml:32]:
```suggestion
          go-version-file: magefiles/go.mod
```

File: .github/workflows/qa-checker.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+name: Code quality checker
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  workflow_dispatch:
```
Comment (huseyinbabal) [inline .github/workflows/qa-checker.yaml:9]: Is this needed?

**Review by huseyinbabal — COMMENTED** [review]
LGTM with minor comments, good job 👍

**Review by huseyinbabal — APPROVED** [review]

**Pull request merged**